### PR TITLE
Make this repo installable with all python versions

### PR DESCRIPTION
**Pull request by winfried-loetzsch**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
List any dependencies that are required for this change.

Fixes # issue

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring including code style reformatting 
- [ ] Other (please describe):

# Checklist:

- [ ] I have read the [contributing guideline doc](https://squirrel-datasets-core.readthedocs.io/en/latest/code_of_conduct.html) (external contributors only)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have kept the PR small so that it can be easily reviewed  
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] All dependency changes have been reflected in the pip requirement files.

### CODE REVIEW

File: requirements.dev.in [hunk 1]
```diff
@@ -15,4 +15,5 @@ notebook>=6.4.12
 jupyter
 networkx
 pre-commit==2.16.0
-pip-tools
\ No newline at end of file
+pip-tools
+torchvision
```
Comment (AlpAribal) [inline requirements.dev.in:19]: There is a `torchvision` flavor as well, do we really need this in dev too?

Comment (AlpAribal) [reply requirements.dev.in:19]: `squirrel_plugin.squirrel_drivers()` may fail without torchvision. I guess we will need a try...except block there

File: src/squirrel_datasets_core/squirrel_plugin.py [hunk 2]
```diff
@@ -11,9 +11,9 @@
 def squirrel_drivers() -> List[Type[Driver]]:
     """Custom drivers added by this package."""
     import squirrel_datasets_core.datasets as ds
+    from squirrel_datasets_core.driver.hub import HubDriver
     from squirrel_datasets_core.driver.huggingface import HuggingfaceDriver
     from squirrel_datasets_core.driver.torchvision import TorchvisionDriver
```
Comment (AlpAribal) [inline src/squirrel_datasets_core/squirrel_plugin.py:16]: I expect this line to fail when torchvision is not installed.

File: examples/02.Catalog.ipynb [hunk 3]
```diff
@@ -87,10 +87,18 @@
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 3,
    "id": "0f67fb90",
    "metadata": {},
-   "outputs": [],
+   "outputs": [
```
Comment (AlpAribal) [inline examples/02.Catalog.ipynb:93]: Could you strip the notebooks as well?

**Review by AlpAribal — APPROVED** [review]
LGTM!

File: examples/01.Getting_Started.ipynb [hunk 4]
```diff
@@ -25,7 +25,7 @@
     "\n",
     "    restart_colab = False\n",
     "except:\n",
-    "    !pip install -q --ignore-requires-python --upgrade squirrel-datasets-core matplotlib # noqa\n",
+    "    !pip install -q --ignore-requires-python --upgrade \"squirrel-datasets-core[torchvision]\" matplotlib # noqa\n",
```
[the remark this answers is not in the document]
Comment (winfried-loetzsch) [reply examples/01.Getting_Started.ipynb:28]: It works both ways

**Pull request merged**